src/consensus/versionbits.h

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2009-2015 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_VERSIONBITS_H
+#define BITCOIN_VERSIONBITS_H
+
+#include <map>
+#include <set>
+#include <stdexcept>
+#include <stdint.h>
+
+namespace Consensus
+{
+
+const int           NO_RULE      = -1;
+
+namespace VersionBits
+{
+
+const int           VERSION_HIGH_BITS   = 0x20000000;
+const int           VERSION_BITS_MASK   = 0x1fffffff;
+const char          MIN_BIT             = 0;
+const char          MAX_BIT             = 28;
+
+enum RuleState { UNDEFINED, DEFINED, LOCKED_IN, ACTIVE, FAILED };
+
+const char* GetRuleStateText(int ruleState, bool bUseCaps = false);
+
+bool UsesVersionBits(int nVersion);
+
+class SoftFork
+{
+public:
+    SoftFork(int bit, int rule, int threshold, uint32_t deployTime, uint32_t expireTime) :
+        m_bit(bit),
+        m_rule(rule),
+        m_threshold(threshold),
+        m_deployTime(deployTime),
+        m_expireTime(expireTime) { }
+
+    int         GetBit() const { return m_bit; }
+    int         GetRule() const { return m_rule; }
+    int         GetThreshold() const { return m_threshold; }
+    uint32_t    GetDeployTime() const { return m_deployTime; }
+    uint32_t    GetExpireTime() const { return m_expireTime; }
+
+private:
+    int m_bit;
+    int m_rule;
+    int m_threshold;
```

Comment:
@rustyrussell We can just use the same threshold for each soft fork in a given chain. In the future if we ever decide to change the threshold, it will be straightforward to do so without requiring any changes to the implementation while allowing us to keep the old thresholds for the earlier forks. The BIP document mentions the possibility of modifying the threshold at some point in the future. So I don't see any downside in supplying this parameter for each fork.

Adding the parameter to Consensus::Params means that if we ever decide to support a different threshold we'll have to hack around it, which seems completely unnecessary when the current implementation can easily support it already (or even potentially support a different soft fork mechanism we haven't come up with yet) without the risk of polluting the Consensus::Params structure any further with legacy stuff (i.e. the ISM stuff now remains in there as legacy...but probably shouldn't be there).